pkg/operation/botanist/cleanup.go

```diff
@@ -266,6 +274,10 @@ func (b *Botanist) CleanKubernetesResources(ctx context.Context) error {
 		cleanResourceFn(ops, c, &corev1.ServiceList{}, ServiceCleanOption, cleanOptions),
 		cleanResourceFn(ops, c, &appsv1.StatefulSetList{}, StatefulSetCleanOption, cleanOptions),
 		cleanResourceFn(ops, c, &corev1.PersistentVolumeClaimList{}, PersistentVolumeClaimCleanOption, cleanOptions),
+		// Cleaning up VolumeSnapshots can take a longer time if many snapshots were taken.
+		// Hence, we only finalize these objects after 1h.
+		cleanResourceFn(ops, c, &volumesnapshotv1beta1.VolumeSnapshotList{}, VolumeSnapshotContentCleanOption, FinalizeAfterOneHour),
+		cleanResourceFn(ops, c, &volumesnapshotv1beta1.VolumeSnapshotContentList{}, VolumeSnapshotContentCleanOption, FinalizeAfterOneHour),
```

Comment:
Can we still respect the `cleanOptions` here, i.e., if somebody overwrote the default grace period with the `shoot.gardener.cloud/cleanup-kubernetes-resources-finalize-grace-period-seconds` annotation?